File: couch/ruby/delete/couch_delete.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
#
#	couch_delete.rb
#
#					Jun/20/2014
#
require 'json'
require 'restclient'
#
# --------------------------------------------------------------------------
puts "*** 開始 ***"
key_in = ARGV[0]
#
puts key_in
#
url_collection = 'http://localhost:5984/nagano'
url_target = url_collection + '/' + key_in
begin
	json_str = RestClient.get url_target
	unit_aa=JSON.parse(json_str)
	if (unit_aa.key?('_rev'))
		puts unit_aa['name']
		url_del =  url_target + '?rev=' + unit_aa['_rev']
		RestClient.delete url_del
	end
rescue => ex
	puts ex.message
end
#
puts	"*** 終了 ***"
#
# --------------------------------------------------------------------------

File: couch/python/delete/couch_delete.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-
#
#	delete/couch_delete.py
#
#					Aug/13/2017
# ----------------------------------------------------------------
import	os
import sys
import string
import json
import requests
#
sys.path.append('../../../common/python_common')
from requests_get import requests_get_proc
from requests_get import requests_delete_proc
#
# ----------------------------------------------------------------
sys.stderr.write("*** 開始 ***\n")
#
key_in = sys.argv[1]
print("%s" % key_in)
#
url_json = 'http://localhost:5984/nagano'
url_key = url_json + '/' + key_in
#
str_buf_aa = requests_get_proc(url_key)
#
print("len (str_buf) = %d\n" % len(str_buf_aa))
#
unit_aa = json.loads(str_buf_aa)
#
if ('_rev' in unit_aa):
	print(unit_aa['_rev'])
	url_del = url_key + '?rev=' + unit_aa['_rev']
	requests_delete_proc(url_del)
else:
	print("*** not exist ***")
	print(unit_aa)
#
sys.stderr.write("*** 終了 ***\n")
# ----------------------------------------------------------------
